src/qtxdg/xdgdesktopfile.cpp

```diff
@@ -1212,7 +1212,17 @@ QStringList XdgDesktopFile::expandExecString(const QStringList& urls) const
         }
 
         // ----------------------------------------------------------
-        result << expandEnvVariables(token);
+        if (token.startsWith(QLatin1Char('\'')) && token.endsWith(QLatin1Char('\'')))
```

Comment:
This doesn't seem correct for me. 
1. consider situation, when the `'` is part of the file name -> it will be stripped wrongly
2. consider situation, when the `%f` would be surrounded by `'` in .desktop (for any reason) -> the `%f` won't be correctly expanded
3. even the `static QStringList parseCombinedArgString(const QString &program)` is wrong and does not consider escapes defined by spec

Reply:
Thanks for the review!

> consider situation, when the ' is part of the file name -> it will be stripped wrongly

It won't because the condition isn't satisfied. A specific example would be very appreciated.

> consider situation, when the %f would be surrounded by '

There's no change of behavior before and after the patch as far as I understand your point. A specific example?